float64 cx
float64 cy
float64 hit
float64 area
